designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala

```diff
@@ -0,0 +1,270 @@
+package pl.touk.nussknacker.ui.api
+
+import io.restassured.RestAssured.given
+import org.scalatest.freespec.AnyFreeSpecLike
+import pl.touk.nussknacker.engine.api.component.{ComponentId, ComponentType}
+import pl.touk.nussknacker.test.{
+  NuRestAssureExtensions,
+  NuRestAssureMatchers,
+  PatientScalaFutures,
+  RestAssuredVerboseLogging
+}
+import pl.touk.nussknacker.ui.api.helpers.{
+  NuItTest,
+  NuScenarioConfigurationHelper,
+  TestProcessingTypes,
+  WithMockableDeploymentManager
+}
+import io.restassured.module.scala.RestAssuredSupport.AddThenToResponse
+import org.hamcrest.Matchers.equalTo
+import org.scalatest.matchers.must.Matchers.contain
+import pl.touk.nussknacker.engine.api.process.ProcessName
+import pl.touk.nussknacker.engine.build.ScenarioBuilder
+import pl.touk.nussknacker.engine.definition.{ComponentIdProvider, DefaultComponentIdProvider}
+import pl.touk.nussknacker.test.ProcessUtils.convertToAnyShouldWrapper
+import pl.touk.nussknacker.ui.api.helpers.TestCategories.Category1
+
+import scala.jdk.CollectionConverters.CollectionHasAsScala
+
+class ComponentResourcesApiSpec
+    extends AnyFreeSpecLike
+    with NuItTest
+    with WithMockableDeploymentManager
+    with NuScenarioConfigurationHelper
+    with NuRestAssureExtensions
+    with NuRestAssureMatchers
+    with RestAssuredVerboseLogging
+    with PatientScalaFutures {
+
+  private val defaultComponentIdProvider: ComponentIdProvider = new DefaultComponentIdProvider(Map.empty)
+
+  "The components endpoint should" - {
+    "return users(test, admin) components list" in {
```

Comment:
I think we should have here at least 2 tests:
1. testing if the service returns components
2. testing if the service returns the same components for `test` and `admin` users